src/Ideas.js

```diff
@@ -8,91 +8,116 @@ import './Ideas.css';
 
-function checkIfAllSelected(categories, all) {
+const allValues = {};
+
+const checkIfAllSelected = (categories, all) => {
   let allSelected = true;
   all.forEach((d) => {
     if (categories.indexOf(d) < 0) allSelected = false;
   });
   return allSelected;
-}
+};
 
 // utility helpers for arrays
-function removeItem(array, value) {
+const removeItem = (array, value) => {
   const index = array.indexOf(value);
 
   if (index > -1) {
     array.splice(index, 1);
   }
 
   return array;
-}
+};
+
+const unique = array => array.filter((x, i, a) => a.indexOf(x) === i);
+
 
 class Ideas extends Component {
   constructor(props) {
     super();
-    const { location } = props;
+    const { location, ideas } = props;
     const query = new URLSearchParams(location.search);
-    const value = query.get('categories') || '';
+    const queryCategories = query.get('categories') || '';
+    const queryTags = query.get('tags') || '';
+
+    allValues.categories = allCategories;
+    allValues.tags = unique(ideas.reduce((a, b) => a.concat(b.tags), []).filter(Boolean));
 
-    const categories = value ? value.split(',') : defaultSelection;
+    const categories = queryCategories ? queryCategories.split(',') : allValues.categories;
+    const tags = queryTags ? queryTags.split(',') : allValues.tags;
 
-    this.state = { categories, search: '' };
+    this.state = {
+      categories,
+      tags,
+      search: '',
+    };
   }
 
   componentWillReceiveProps(nextProps) {
     const queryParams = new URLSearchParams(nextProps.location.search);
 
     if (!queryParams.get('categories')) {
-      this.setState({
-        categories: defaultSelection,
-      });
+      this.setState({ categories: allValues.categories });
     }
   }
 
   changeSearch = (event) => {
     this.setState({ search: event.target.value });
   }
 
-  changeCategory = (clickedCategory) => {
-    let categories = this.state.categories.slice();
-    const { history } = this.props;
+  // updates filter state and query params based on a
+  // clicked value, list of all possible values, and type
+  changeFilter = (value, all, type) => {
+    let currentSelected = this.state[type].slice();
 
-    const allWereSelected = checkIfAllSelected(categories, defaultSelection);
+    const allWereSelected = checkIfAllSelected(currentSelected, all);
 
     if (allWereSelected) {
-      categories = [clickedCategory];
-    } else if (categories.indexOf(clickedCategory) > -1) {
-      categories = removeItem(categories, clickedCategory);
+      currentSelected = [value];
+    } else if (currentSelected.indexOf(value) > -1) {
+      currentSelected = removeItem(currentSelected, value);
     } else {
-      categories.push(clickedCategory);
+      currentSelected.push(value);
     }
 
-    if (categories.length === 0) categories = defaultSelection;
+    if (currentSelected.length === 0) currentSelected = all;
+
+    this.state[type] = currentSelected;
```

Comment:
Browser warning: `Do not mutate state directly. Use setState().`

Not sure how it affects things in this particular case (it's working!!), but I think it's a best practice.